rbac/management/tenant_service/v2.py

```diff
@@ -221,7 +222,11 @@ def _disable_user_in_tenant(self, user: User):
 
             for group in principal.group.all():
                 group.principals.remove(principal)
-                tuples_to_remove.append(group.relationship_to_principal(principal))
+                tupple = group.relationship_to_principal(principal)
+                if tupple is None:
+                    error_message = f"principal {principal.username} has no user id."
+                else:
+                    tuples_to_remove.append(tupple)
```

Comment:
Trying to think through our options here... This is really an illegal state – we should never get to the point where we're trying to remove a user that we don't have an ID for. We could...

1. Delete the Principal even if we can't remove the tuple. This would keep us in an illegal state, where the graph is inconsistent with RBAC. But at least we would have removed the user from RBAC's v1 access result.
2. Keep the Principal. This keeps the access graph consistent with RBAC, but leaves the user around even though we know it should be removed.

I'm wondering if 1 is really better? What is the impact of leaving the principal? What is the impact of deleting it?

For example, there may be some code that assumes if a Principal isn't there, then it's already been removed from the graph. Actually I think there's an example of that right here in this method:

```python
        except Principal.DoesNotExist:
            pass
```

This would actually make it hard for us to repair the data, because we'd no longer be able to rely on a message correcting it, we'd have to manually fix the graph.

We could change this to try to remove the tuple from the graph anyway, but that has its own side effect of adding redundant work in the happy path.

I'm wondering if it's better to raise an exception early after all? What do you think?